### ISSUE TITLE
meta.dart does not compile on sdk 1.24.3

### ISSUE THREAD

**Issue by rmbarnett-rice** [dart-lang/sdk#42935]
* Dart SDK Version: 1.24.3
* Using Windows

```
Compiling askbayoucom|web/main.dart...
[Error from Dart2JS on askbayoucom|web/main.dart]:
web\packages\meta\meta.dart:50:36:
Not a compile-time constant.
const _AlwaysThrows alwaysThrows = _AlwaysThrows();
                                   ^^^^^^^^^^^^^^^
```

**Comment by srawlins** [comment]
I suspect that upgrading from Dart 1.24.3 is going to be your best bet. Dart 1.24.3 is not a supported product.

**Comment by mraleph** [comment]
I think the actual problem is that we released package meta with incorrect SDK constraints: all new versions (1.2.*) have incorrect constraint `>=1.12.0` while using Dart 2 syntax (implicit const). 

I don't know what the procedure here should be though. I'd say we need to purge these versions and release one with correct SDK constraint. 

/cc @bwilkerson

**Comment by kevmoo** [comment]
It does seem that the issue in the SDK constraint on meta.

As with similar issues, what to do next doesn't leave a lot of easy options.

**Comment by lrhn** [comment]
See: https://stackoverflow.com/questions/63252769/not-a-compile-time-constant-const-alwaysthrows-alwaysthrows-alwaysthrows/63279229#63279229

**Comment by jakemac53** [comment]
Note that we cannot remove these versions from pub because lots of users already have them - in general packages should never be removed because that breaks users that are currently using that version (either pinning it directly, checking in their pubspec.lock files, or running `pub get` after purging their pub cache).

**Comment by jakemac53** [comment]
My suggestion would be to tell users to pin to an old meta version, this has been an issue for a long time but doesn't seem to be affecting very many people so it isn't worth trying to resolve, imo.

**Comment by mraleph** [comment]
Can't we just republish the same versions but update their `pubspecs.yaml`? I'd assume we have power to write over corresponding cloud storage buckets /cc @jonasfj

**Comment by jakemac53** [comment]
The philosophy has always been not to do this - package versions are generally guaranteed not to change and overwriting them is not something we have ever allowed. It sets a pretty dangerous precedent and can lead to unforeseen issues.

For instance for a long time (and maybe still today) flutter shipped with its own zip file containing pre-downloaded versions of its packages, which it would use to prime the pub cache effectively. That is arguably a dangerous thing to do, but they at least did do it before, and is just one example of a way you might run into problems.

A more simple example is that many users already have these versions in their pub caches, so those would never be updated to the replaced version unless they explicitly purged their cache.

**Comment by sigmundch** [comment]
In this case, maybe it would be simpler to change the code to use explicit consts and republish as a patch release, then the original version range is still valid?

If that's not enough, I recall we don't update existing versions, but I thought we often release two new releases to overshadow the previous releases that have invalid constraints. So an unconstrained `pub get` will get a version that works.  For example:

* release 1.2.2+1 which makes a breaking change to revert back to 1.8.*, but has a strict version range: ">=1.2 <2.0.0"
* release 1.2.2+2 which matches 1.2.2, but has a correct lowerbound: ">= 2.0.0"

**Comment by jakemac53** [comment]
Yes a patch release would be one valid way to try and resolve this issue.

But, this is something that has been a potential problem for a very long time with no reports. I really don't think it is worth fixing, very few people have ran into it and they have a workaround (pin meta).

**Comment by rmbarnett-rice** [comment]
> 
> 
> See: https://stackoverflow.com/questions/63252769/not-a-compile-time-constant-const-alwaysthrows-alwaysthrows-alwaysthrows/63279229#63279229

I'm the author of that post as well.

**Comment by bwilkerson** [comment]
I'm going to close this under the assumption that we have resolved the issue by now, if for no other reason that because of having published newer versions of `meta`.
